src/common/xo/index.js

```diff
@@ -1328,6 +1331,53 @@ export const changePassword = (oldPassword, newPassword) => (
   )
 )
 
+const _mergeUserPreferences = (user, preferences) => (
+  _call('user.set', {
+    id: resolveId(user),
+    preferences: {
+      ...user.preferences,
+      ...preferences
+    }
+  })::tap(
+    subscribeCurrentUser.forceRefresh
+  )
+)
```

Comment:
You don't have to merge the new entries with the existing preferences yourself, the server do that automatically.